### DISCUSSION TITLE
Error when composing on server

### DISCUSSION THREAD

**Discussion by tryggvekj** [INGInious/INGInious#1005] [Q&A]
I tried composing on a server, but when I did, I got this error:
agent-docker-1  | Traceback (most recent call last):
agent-docker-1  |   File "/usr/local/lib/python3.9/site-packages/requests/adapters.py", line 633, in send
agent-docker-1  |     conn = self.get_connection_with_tls_context(
agent-docker-1  |   File "/usr/local/lib/python3.9/site-packages/requests/adapters.py", line 489, in get_connection_with_tls_context
agent-docker-1  |     conn = self.poolmanager.connection_from_host(
agent-docker-1  |   File "/usr/local/lib/python3.9/site-packages/urllib3/poolmanager.py", line 303, in connection_from_host
agent-docker-1  |     return self.connection_from_context(request_context)
agent-docker-1  |   File "/usr/local/lib/python3.9/site-packages/urllib3/poolmanager.py", line 325, in connection_from_context
agent-docker-1  |     raise URLSchemeUnknown(scheme)
agent-docker-1  | urllib3.exceptions.URLSchemeUnknown: Not supported URL scheme http+docker
agent-docker-1  |
agent-docker-1  | During handling of the above exception, another exception occurred:
agent-docker-1  |
agent-docker-1  | Traceback (most recent call last):
agent-docker-1  |   File "/usr/local/lib/python3.9/site-packages/docker/api/client.py", line 213, in _retrieve_server_version
agent-docker-1  |     return self.version(api_version=False)["ApiVersion"]
agent-docker-1  |   File "/usr/local/lib/python3.9/site-packages/docker/api/daemon.py", line 181, in version
agent-docker-1  |     return self._result(self._get(url), json=True)
agent-docker-1  |   File "/usr/local/lib/python3.9/site-packages/docker/utils/decorators.py", line 44, in inner
agent-docker-1  |     return f(self, *args, **kwargs)
agent-docker-1  |   File "/usr/local/lib/python3.9/site-packages/docker/api/client.py", line 236, in _get
agent-docker-1  |     return self.get(url, **self._set_request_timeout(kwargs))
agent-docker-1  |   File "/usr/local/lib/python3.9/site-packages/requests/sessions.py", line 602, in get
agent-docker-1  |     return self.request("GET", url, **kwargs)
agent-docker-1  |   File "/usr/local/lib/python3.9/site-packages/requests/sessions.py", line 589, in request
agent-docker-1  |     resp = self.send(prep, **send_kwargs)
agent-docker-1  |   File "/usr/local/lib/python3.9/site-packages/requests/sessions.py", line 703, in send
agent-docker-1  |     r = adapter.send(request, **kwargs)
agent-docker-1  |   File "/usr/local/lib/python3.9/site-packages/requests/adapters.py", line 637, in send
agent-docker-1  |     raise InvalidURL(e, request=request)
agent-docker-1  | requests.exceptions.InvalidURL: Not supported URL scheme http+docker
agent-docker-1  |
agent-docker-1  | The above exception was the direct cause of the following exception:
agent-docker-1  |
agent-docker-1  | Traceback (most recent call last):
agent-docker-1  |   File "/inginious/inginious-agent-docker", line 130, in <module>
agent-docker-1  |     loop.run_until_complete(agent.run())
agent-docker-1  |   File "/usr/lib64/python3.9/asyncio/base_events.py", line 647, in run_until_complete
agent-docker-1  |     return future.result()
agent-docker-1  |   File "/inginious/inginious/agent/docker_agent/__init__.py", line 968, in run
agent-docker-1  |     await self._init_clean()
agent-docker-1  |   File "/inginious/inginious/agent/docker_agent/__init__.py", line 131, in _init_clean
agent-docker-1  |     self._runtimes = self._detect_runtimes()
agent-docker-1  |   File "/inginious/inginious/agent/docker_agent/__init__.py", line 993, in _detect_runtimes
agent-docker-1  |     for runtime in self._docker.sync.list_runtimes().keys():
agent-docker-1  |   File "/inginious/inginious/agent/docker_agent/_docker_interface.py", line 276, in list_runtimes
agent-docker-1  |     return {name: x["path"] for name, x in self._docker.info()["Runtimes"].items()}
agent-docker-1  |   File "/inginious/inginious/agent/docker_agent/_docker_interface.py", line 32, in _docker
agent-docker-1  |     return docker.from_env()
agent-docker-1  |   File "/usr/local/lib/python3.9/site-packages/docker/client.py", line 94, in from_env
agent-docker-1  |     return cls(
agent-docker-1  |   File "/usr/local/lib/python3.9/site-packages/docker/client.py", line 45, in __init__
agent-docker-1  |     self.api = APIClient(*args, **kwargs)
agent-docker-1  |   File "/usr/local/lib/python3.9/site-packages/docker/api/client.py", line 197, in __init__
agent-docker-1  |     self._version = self._retrieve_server_version()
agent-docker-1  |   File "/usr/local/lib/python3.9/site-packages/docker/api/client.py", line 220, in _retrieve_server_version
agent-docker-1  |     raise DockerException(
agent-docker-1  | docker.errors.DockerException: Error while fetching server API version: Not supported URL scheme http+docker
agent-docker-1  | 2024-07-02 15:19:37,549 - inginious - INFO - Closing loop
agent-docker-1  | 2024-07-02 15:19:37,549 - inginious - INFO - Waiting for ZMQ to send remaining messages to backend (can take 1 sec)
agent-docker-1  | 2024-07-02 15:19:37,549 - inginious - INFO - Done
agent-docker-1  | 2024-07-02 15:19:37,554 - inginious - ERROR - Agent has received an exception forcing it to restart

And then the same exact error gets called over and over again. Does anybody have a clue on what's going on?

**Comment by nrybowski** [comment]
@Tryggvekj is this still an issue?
